pkg/registry/generic/etcd/etcd.go

```diff
@@ -398,6 +398,11 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	return e.finalizeDelete(out, true)
 }
 
+// TODO Should this be here, or should each resource have it's own DeleteCollection()?
+func (e *Etcd) DeleteCollection(ctx api.Context) (runtime.Object, error) {
```

Comment:
Or for resources that want to support this pattern, apply an adapter that I can use to decorate the registry with this feature, and we would add a unique behavior at that point.